packages/clay-autocomplete/src/DataProvider.js

```diff
@@ -0,0 +1,58 @@
+import {isString} from 'metal';
+import fuzzy from 'fuzzy';
+import EventEmitter from 'metal-events';
+
+/* eslint-disable require-jsdoc */
+
+function getDataFromArray(query, data, cb) {
+	cb(
+		fuzzy
+			.filter(query, data, {
+				// pre: '<strong>',
+				// post: '</strong>',
+				extract: el => el.label,
+			})
+			.map(e => ({label: e.string}))
+	);
+}
+
+function getDataFromURL(query, url, paramName = 'q', cb) {
+	fetch(`${url}?${paramName}=${query}`)
```

Comment:
We need to manage all kind of urls, for example, what if the passed url already has queryParams? You can use `metal-uri` https://github.com/metal/metal-plugins/tree/master/packages/metal-uri to deal with this.